lib/src/models/ble_device.dart

```diff
@@ -1,30 +1,36 @@
 import 'dart:typed_data';
 
-class BleScanResult {
+import 'package:universal_ble/universal_ble.dart';
+
+class BleDevice {
   String deviceId;
   String? name;
+  int? rssi;
   bool? isPaired;
+  List<String> services;
+  bool? isSystemDevice;
   Uint8List? manufacturerDataHead;
   Uint8List? manufacturerData;
-  int? rssi;
-  List<String> services;
 
-  BleScanResult({
-    required this.name,
+  Future<bool> get isConnected => UniversalBle.isConnected(deviceId);
```

Comment:
Maybe if we also want to track the connecting/disconnecting status we should convert that to an enum similar to [CoreBluetooth](https://developer.apple.com/documentation/corebluetooth/cbperipheralstate).